tests/contrib/gevent/test_monkeypatch.py

```diff
@@ -20,17 +20,22 @@ def test_gevent_warning(monkeypatch):
     assert b"RuntimeWarning: Loading ddtrace before using gevent monkey patching" in subp.stderr.read()
 
 
-@pytest.mark.subprocess
-def test_gevent_auto_patching():
```

Comment:
This refactor prevents this test from logging ``Tracer was shutdown. Spans generated after a tracer is shutdown will not be sent.``

Reply:
I am confused by this comment, the changes you made have it where this test does *not* log this warning, and therefore it fails?

Reply:
The ``pyest.mark.subprocess`` annotation calles [run_function_from_file](https://github.com/DataDog/dd-trace-py/blob/1.x/tests/conftest.py#L160) and then fails the assertion on [line 197](https://github.com/DataDog/dd-trace-py/blob/1.x/tests/conftest.py#L160) after checking the output of ``ddtrace-run test.py``. I couldn't reproduce this error locally but here's the error in [ci](https://app.circleci.com/pipelines/github/DataDog/dd-trace-py/15672/workflows/e8cf1079-1d80-4423-be99-8e38f547be84/jobs/1080110). Using ``run_python_code_in_subprocess or ddtrace_run_python_code_in_subprocess`` fixtures instead prevented this warning log (Now the AgentWriter is not shutdown before tracer.trace()) is called. I don't think patching or accessing the ``GeventContextProvider`` should produce traces so I have no idea why this test is failing.